mm2src/mm2_bitcoin/chain/src/block_header.rs

```diff
@@ -2554,4 +2556,79 @@ mod tests {
+            224, 121, 220, 228, 240, 91, 93, 188, 24, 194, 168, 114, 178, 37, 9, 116, 14, 190, 106, 57, 3, 224, 10,
+            209, 171, 197, 80, 118, 68, 24, 98, 100, 63, 147, 96, 110, 61, 195, 94, 141, 159, 44, 174, 243, 238, 107,
+            225, 77, 81, 59, 46, 6, 43, 33, 208, 6, 29, 227, 189, 86, 136, 23, 19, 161, 165, 193, 127, 90, 206, 5, 225,
+            236, 9, 218, 83, 249, 148, 66, 223, 23, 90, 73, 189, 21, 74, 169, 110, 73, 73, 222, 205, 82, 254, 215, 156,
+            207, 124, 203, 206, 50, 148, 20, 25, 195, 20, 227, 116, 228, 163, 150, 172, 85, 62, 23, 181, 52, 3, 54,
+            161, 162, 92, 34, 249, 228, 42, 36, 59, 165, 64, 68, 80, 182, 80, 172, 252, 130, 106, 110, 67, 41, 113,
+            172, 231, 118, 225, 87, 25, 81, 94, 22, 52, 206, 185, 164, 163, 80, 97, 182, 104, 199, 73, 152, 211, 223,
+            181, 130, 127, 98, 56, 236, 1, 83, 119, 230, 249, 201, 79, 56, 16, 135, 104, 207, 110, 92, 139, 19, 46, 3,
+            3, 251, 90, 32, 3, 104, 248, 69, 173, 157, 70, 52, 48, 53, 166, 255, 148, 3, 29, 248, 216, 48, 148, 21,
+            187, 63, 108, 213, 237, 233, 193, 53, 253, 171, 204, 3, 5, 153, 133, 141, 128, 60, 15, 133, 190, 118, 97,
+            200, 137, 132, 216, 143, 170, 61, 38, 251, 14, 154, 172, 0, 86, 165, 63, 27, 93, 11, 174, 215, 19, 200, 83,
+            196, 162, 114, 104, 105, 160, 161, 36, 168, 165, 187, 192, 252, 14, 248, 12, 138, 228, 203, 83, 99, 106,
+            160, 37, 3, 184, 106, 30, 185, 131, 111, 204, 37, 152, 35, 226, 105, 45, 146, 29, 136, 225, 255, 193, 230,
+            203, 43, 222, 67, 147, 156, 235, 63, 50, 166, 17, 104, 111, 83, 159, 143, 124, 159, 11, 240, 3, 129, 247,
+            67, 96, 125, 64, 150, 15, 6, 211, 71, 209, 205, 138, 200, 165, 25, 105, 194, 94, 55, 21, 14, 253, 247, 170,
+            76, 32, 55, 162, 253, 5, 22, 251, 68, 69, 37, 171, 21, 122, 14, 208, 167, 65, 43, 47, 166, 155, 33, 127,
+            227, 151, 38, 49, 83, 120, 44, 15, 100, 53, 31, 189, 242, 103, 143, 160, 220, 133, 105, 145, 45, 205, 142,
+            60, 202, 211, 143, 52, 242, 59, 187, 206, 20, 198, 162, 106, 194, 73, 17, 179, 8, 184, 44, 126, 67, 6, 45,
+            24, 11, 174, 172, 75, 167, 21, 56, 88, 54, 92, 114, 198, 61, 207, 95, 106, 91, 8, 7, 11, 115, 10, 219, 1,
+            122, 234, 233, 37, 183, 208, 67, 153, 121, 226, 103, 159, 69, 237, 47, 37, 167, 237, 207, 210, 251, 119,
+            168, 121, 70, 48, 40, 92, 203, 10, 7, 31, 92, 206, 65, 11, 70, 219, 249, 117, 11, 3, 84, 170, 232, 182, 85,
+            116, 80, 28, 198, 158, 251, 91, 106, 67, 68, 64, 116, 254, 225, 22, 100, 27, 178, 157, 165, 108, 43, 74,
+            127, 69, 105, 145, 252, 146, 178,
+        ];
+        let expected_hash = dhash256(serialized_headers.as_slice());
+        let header: BlockHeader = deserialize(serialized_headers.as_slice()).unwrap();
+
+        assert_eq!(expected_hash, header.hash());
```

Comment:
Why not compare the actual headers like other tests instead of comparing hashes?

Reply:
done